tests/examples/test_examples.py

```diff
@@ -62,31 +68,38 @@ def test_file_examples(file_example, example, report):
         else:
             _get_pdiff(example)
 
-
-### {{{ THIS IS BROKEN and all examples are skipped in examples.yaml
 @pytest.mark.examples
-def test_server_examples(server_example, example, bokeh_server, report):
+def test_server_examples(server_example, example, webdriver, bokeh_server, report):
     if example.is_skip:
         pytest.skip("skipping %s" % example.relpath)
 
-    # Note this is currently broken - server uses random sessions but we're
-    # calling for "default" here - this has been broken for a while.
-    # https://github.com/bokeh/bokeh/issues/3897
-    url = '%s/?bokeh-session-id=%s' % (bokeh_server, example.name)
-    assert _run_example(example) == 0, 'Example did not run'
+    app = build_single_handler_application(example.path)
+    doc = app.create_document()
 
-    if example.no_js:
-        if not pytest.config.option.no_js:
-            warn("skipping bokehjs for %s" % example.relpath)
-    else:
-        _assert_snapshot(example, url, 'server')
+    session_id = "session_id"
+    push_session(doc, session_id=session_id)
 
-        if example.no_diff:
-            warn("skipping image diff for %s" % example.relpath)
-        else:
-            _get_pdiff(example)
+    webdriver.get("http://localhost:5006/?bokeh-session-id=%s" % session_id)
+    webdriver.set_window_size(1200, 800)
+
+    _wait_until_render_complete(webdriver, timeout=20)
+
+    assert (has_no_console_errors(webdriver))
```

Comment:
This is cruft and should be removed. ``_wait_until_render_complete`` checks the browser logs already, though it might be better if that's broken out.